src/Sylius/Bundle/ApiBundle/EventListener/ApiCartBlamerListener.php

```diff
@@ -45,7 +45,7 @@ public function onInteractiveLogin(InteractiveLoginEvent $interactiveLoginEvent)
         }
 
         $cart = $this->getCart();
-        if (null === $cart || null !== $cart->getCustomer()) {
+        if (null === $cart || !$cart->getCreatedByGuest()) {
```

Comment:
That change assumes that the flag would **always** be set when the customer is set as well. So it only bumps the need of some custom method on the `Order` entity (as mentioned in one of the above comment 🚀)

Reply:
In this case, it makes total sense, because of the order wasn't already claimed, now it will be. But for sure custom method is viable option for setting it.